.travis.yml

```diff
@@ -32,74 +32,81 @@ addons:
          - os: linux
-           env: TEST_MODE='doctest'
+           env: JOB="Py2Doctest" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -b doctest' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 # doctest will fail till new version of astropy-helpers
          - os: linux
-           env: PYTHON_VERSION=3.4 TEST_MODE='python3'
+           env: JOB="Py2NumpyPrev" PYTHON_VERSION=2.7 SETUP_CMD='test' NUMPY_VERSION=1.9.2 CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+         - os: linux
+           env: JOB="Py2Figures" PYTHON_VERSION=2.7 SETUP_CMD='test --figure' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+         - os: linux
+           env: JOB="Py3" PYTHON_VERSION=3.4 MAIN_CMD='continuous-integration/python3.sh' SETUP_CMD='' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
 
-# If the online build fails, we don't want it to report the whole build broken.
+    # allow_failures has to repeat the environment from the matrix above to mark it as such
     allow_failures:
-        - env: TEST_MODE='online'
-        - env: TEST_MODE='doctest'
-
-# Install miniconda
-before_install:
-    - source continuous-integration/travis/install_$TRAVIS_OS_NAME.sh
+         - env: JOB="Py2Doctest" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -b doctest' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 # doctest will fail till new version of astropy-helpers
+         - env: JOB="Py2Coverage" PYTHON_VERSION=2.7 SETUP_CMD='test --online --coverage --cov-report=html' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 PIP_DEPENDENCIES='pytest-cov'
+         - env: JOB="Py2Figures" PYTHON_VERSION=2.7 SETUP_CMD='test --figure' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
 
 install:
-    # Add a conda env and set up extra binstar channels
-    - conda create --yes -n test python=$PYTHON_VERSION
-    - conda config --add channels sunpy --add channels astropy --add channels astropy-ci-extras --add channels https://conda.binstar.org/sunpy/channel/citesting
-    - source activate test
-    # Install default dependencies
-    - conda install --yes pip
-    - conda install -q --yes numpy$NUMPY_VERSION pandas$PANDAS_VERSION scipy matplotlib$MATPLOTLIB_VERSION requests beautiful-soup sqlalchemy scikit-image pytest jinja2 cython wcsaxes
-    - pip install pytest-cov coverage coveralls
-    # Install sunpy conda packages
-    - conda install -q --yes numpy=$NUMPY_VERSION suds-jurko glymur
-    # Install the correct version of astropy
-    - if [[ $TEST_MODE != 'astropy-dev' ]]; then conda install -q --yes astropy>=0.4.0 numpy$NUMPY_VERSION; fi
-    - if [[ $TEST_MODE == 'astropy-dev' ]]; then pip install git+git://github.com/astropy/astropy.git; fi
-    # Install sphinx if needed
-    - if [[ $TEST_MODE == 'sphinx' || $TEST_MODE == 'doctest' ]]; then conda install -q --yes sphinx numpy$NUMPY_VERSION; fi
-    - if [[ $TEST_MODE == 'sphinx' || $TEST_MODE == 'doctest' ]]; then python -c "import sunpy.data; sunpy.data.download_sample_data()"; fi
+    # install openjpeg
```

Comment:
I take it all back, we should get this working because we are skipping the Glymur tests: https://travis-ci.org/sunpy/sunpy/jobs/105999324#L795